core/src/main/java/io/ddf/datasource/DataFormat.java

```diff
@@ -0,0 +1,15 @@
+package io.ddf.datasource;
+
+
+/**
+ */
+
+public enum DataFormat {
+  UNDEF(-1), SQL(0), CSV(1), TSV(2), JSON(3), PARQUET(4);
+
+  private int value;
+
+  private DataFormat(int value) {
+    this.value = value;
+  }
```

Comment:
The above conversion has to exist somewhere, and I think it fits here. Also we should leave UNDEF for default case and not limit to -1, since what if some weird number got in and it's not 0-4 or -1.